src/alphabets/mod.rs

```diff
@@ -152,6 +152,72 @@ impl Alphabet {
     pub fn is_empty(&self) -> bool {
         self.symbols.is_empty()
     }
+
+    /// Return a new alphabet taking the intersect between this and other.
+    ///
+    /// # Example
+    /// ```
+    /// use bio::alphabets;
+    ///
+    /// let alpha_a = alphabets::Alphabet::new(b"acgtACGT");
+    /// let alpha_b = alphabets::Alphabet::new(b"atcgMVP");
+    /// let intersect_alpha = alpha_a.intersection(alpha_b);
+    ///
+    /// assert_eq!(intersect_alpha, alphabets::Alphabet::new(b"atcg"));
+    /// ```
+    pub fn intersection(self, other: Alphabet) -> Self {
```

Comment:
I'm not sure if we really want to consume the alphabets used here; i.e. in
```rust
let intersection = some_alphabet.intersection(other_alphabet);
```
neither `some_alphabet` nor `other_alphabet` will be available after the call to `intersection`.

I think I'd prefer to just use references here instead, i.e.:
```suggestion
    pub fn intersection(&self, other: &Alphabet) -> Self {
```

Reply:
👍 -- also in 01cf874